staging/src/k8s.io/apiserver/pkg/storage/util.go

```diff
@@ -135,7 +136,11 @@ func AnnotateInitialEventsEndBookmark(obj runtime.Object) error {
 		objAnnotations = map[string]string{}
 	}
 	objAnnotations[metav1.InitialEventsAnnotationKey] = "true"
+	if unstructuredList, isUnstructured := listObj.(*unstructured.UnstructuredList); isUnstructured {
```

Comment:
could be set for all types.

Reply:
Sharing comments from async conversation here. CRs have per-version storage and their NewList populates the GVK because Unstructured objects must describe their own type (https://github.com/kubernetes/kubernetes/blob/3adafc6a502fa7ab71a74099a43ccf74be46b2a0/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go#L53-L58). For other types NewList typically returns a pointer to a zero-value of the internal version's list type, so the fields of TypeMeta will be empty. Those cases (as I understand it) need a typer to be able to go from Go type to internal GVK, and a converter to go from internal GVK to the response GVK.